src/engine/TPA.cc

```diff
@@ -609,7 +609,8 @@ TPASplit::QueryResult TPASplit::reachabilityQueryExact(PTRef from, PTRef to, uns
 //                std::cout << std::endl;
                 TRACE(3, "Learning " << itp.x)
                 TRACE(4, "Learning " << logic.pp(itp))
-                assert(itp != logic.getTerm_true());
+                // If itp == logic.getTerm_true, then the error states were trivially unreachable
+                if (itp == logic.getTerm_true()) { assert(power == 0); }
```

Comment:
Here, I would keep the original assert. I believe we should not get here when the system is trivially safe.